src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -51,8 +51,13 @@ public BodyModelBinder()
             {
                 var model = await formatter.ReadAsync(formatterContext);
 
-                // key is empty to ensure that the model name is not used as a prefix for validation.
-                return new ModelBindingResult(model, key: string.Empty, isModelSet: true);
+                var isTopLevelObject = bindingContext.ModelMetadata.ContainerType == null;
```

Comment:
Yes, however that is a corner scenario and we should address this when we address that issue.